commit 0820b3173aff4f0f3c803a9e75e726024da38ee5
Author: David Faure <faure@kde.org>
Date:   Thu Dec 6 11:55:05 2012 +0100

    Fix crash when no service was selected (user clicked on "Open With...")

diff --git a/kparts/browserrun.cpp b/kparts/browserrun.cpp
index 6d84800..6de0380 100644
--- a/kparts/browserrun.cpp
+++ b/kparts/browserrun.cpp
@@ -304,7 +304,7 @@ BrowserRun::NonEmbeddableResult BrowserRun::handleNonEmbeddable(const QString& _
                          this, SLOT(slotCopyToTempFileResult(KJob*)) );
                 return Delayed; // We'll continue after the job has finished
             }
-            if (selectedService) {
+            if (selectedService && question.selectedService()) {
                 *selectedService = question.selectedService();
                 // KRun will use this when starting an app
                 KRun::setPreferredService(question.selectedService()->desktopEntryName());
